ed_sensor_integration/src/laser/laser_plugin.cpp

```diff
@@ -546,151 +477,19 @@ void LaserPlugin::update(const ed::WorldModel& world, const sensor_msgs::LaserSc
-        {
-            const ed::EntityConstPtr& e = *e_it;
-            if (e->shape() || !e->has_pose())
-                continue;
-
-            const geo::Pose3D& entity_pose = e->pose();
-            const ed::ConvexHull& entity_chull = e->convexHull();
-
-            if (entity_chull.points.empty())
-                continue;
-
-            //            if (e->existenceProbability() < 0.5 && scan_msg_->header.stamp.toSec() - e->lastUpdateTimestamp() > 1.0) // TODO: magic numbers
-            //            {
-            //                req.removeEntity(e->id());
-            //                continue;
-            //            }
-
-            if (entity_pose.t.x < area_min.x || entity_pose.t.x > area_max.x
-                    || entity_pose.t.y < area_min.y || entity_pose.t.y > area_max.y)
-                continue;
-
-            entities.push_back(e);
-        }
-    }
-
-    // Create association matrix
-    ed_sensor_integration::AssociationMatrix assoc_matrix(clusters.size());
-    for (unsigned int i_cluster = 0; i_cluster < clusters.size(); ++i_cluster)
-    {
-        const EntityUpdate& cluster = clusters[i_cluster];
-
-        for (unsigned int i_entity = 0; i_entity < entities.size(); ++i_entity)
-        {
-            const ed::EntityConstPtr& e = entities[i_entity];
-
-            const geo::Pose3D& entity_pose = e->pose();
-            const ed::ConvexHull& entity_chull = e->convexHull();
-
-            float dx = entity_pose.t.x - cluster.pose.t.x;
-            float dy = entity_pose.t.y - cluster.pose.t.y;
-            float dz = 0;
-
-            if (entity_chull.z_max + entity_pose.t.z < cluster.chull.z_min + cluster.pose.t.z
-                    || cluster.chull.z_max + cluster.pose.t.z < entity_chull.z_min + entity_pose.t.z)
-                // The convex hulls are non-overlapping in z
-                dz = entity_pose.t.z - cluster.pose.t.z;
-
-            float dist_sq = (dx * dx + dy * dy + dz * dz);
-
-            // TODO: better prob calculation
-            double prob = 1.0 / (1.0 + 100 * dist_sq);
-
-            double dt = scan->header.stamp.toSec() - e->lastUpdateTimestamp();
-
-            double e_max_dist = std::max(0.2, std::min(0.5, dt * 10));
-
-            if (dist_sq > e_max_dist * e_max_dist)
-                prob = 0;
-
-            if (prob > 0)
-                assoc_matrix.setEntry(i_cluster, i_entity, prob);
-        }
-    }
-
+    std::vector<ed::EntityConstPtr> entities = findNearbyEntities(clusters, world);
     ed_sensor_integration::Assignment assig;
-    if (!assoc_matrix.calculateBestAssignment(assig))
+    if (!associateSegmentsWithEntities(clusters, entities, scan->header.stamp.toSec(), assig))
     {
         std::cout << "WARNING: Association failed!" << std::endl;
         return;
     }
 
-    std::vector<int> entities_associated(entities.size(), -1);
+    //std::vector<int> entities_associated(entities.size(), -1); // used to clear unaccociated entities
```

Comment:
Why is this commented? If not necessary, please remove

Reply:
It is part of the code to clear unacociated entities, which might be salvagable code. Therefore I wanted to keep it for now